src/main/java/teammates/storage/api/CoursesDb.java

```diff
@@ -153,4 +165,14 @@ private Course getCourseEntity(String courseId) {
     
         return courseList.get(0);
     }
+    
+    private List<Course> getCourseEntities(List<String> courseIds) {
+        Query q = getPM().newQuery(Course.class);
+        q.setFilter(":p.contains(ID)");
+
+        @SuppressWarnings("unchecked")
+        List<Course> courses = (List<Course>) q.execute(courseIds);
+
+        return courses;
+    }
```

Comment:
![batchquery](https://cloud.githubusercontent.com/assets/7261051/8050088/15d1099e-0e9a-11e5-8aee-bbe29f116931.png)
Top = querying one by one, bottom = batch query.
Batch query saves 140 micropennies per course but adding 70 micropennies per course to the batch get. Overall, still saving 70 micropennies per course.